test/ReactWrapper-spec.jsx

```diff
@@ -968,6 +968,23 @@ describeWithDOM('mount', () => {
       expect(setInvalidProps).to.throw(TypeError, similarException.message);
     });
 
+    it('should call the callback when setProps has completed', () => {
+      class Foo extends React.Component {
+        render() {
+          return (
+            <div className={this.props.id}>
+              {this.props.id}
+            </div>
+          );
+        }
+      }
+      const wrapper = mount(<Foo id="foo" />);
+      expect(wrapper.find('.foo').length).to.equal(1);
+      wrapper.setProps({ id: 'bar', foo: 'bla' }, () => {
+        expect(wrapper.find('.bar').length).to.equal(1);
+      });
```

Comment:
I'm not sure usage patterns that setState does act async in testing. but I think you can use `batchedUpdates`  like this.

```js
import { batchedUpdates } from '../src/react-compat';

batchedUpdates(() => {
  wrapper.setProps({ id: 'bar', foo: 'bla' }, () => {
    expect(wrapper.find('.bar').length).to.equal(1);
  });
  expect(wrapper.find('.bar').length).to.equal(0);
});
```